environment.rst

```diff
@@ -0,0 +1,81 @@
+
+
+Recommended code editors
+------------------------
+
+This section provides a few recommendations on code editors you may want to consider. Some of them are included in the CS machines, but you can also install them on your own computer to write your code there (and then follow the steps found in the :ref:`working-remotely` page to compile and eun your code on a CS machine). All of these tools are available on Windows, Mac, and Linux.
+
+You may find that some people are very opinionated about their choice of editor, and will sometimes `argue passionately <https://en.wikipedia.org/wiki/Editor_war>`__ about why their editor is right and yours is wrong. You should ignore these people and ultimately use whatever editor works best for *you* (and don't judge other people for their choice of editor!)
+
+.. figure:: _static/editors.jpg
+   :align: center
+   :alt: Use whatever editor works for you
+
+   *Design by Borja Sotomayor, artwork by Sarah Becan*
+
+Terminal-based editors
+~~~~~~~~~~~~~~~~~~~~~~
+
+Terminal-based editors, like `vim <https://www.vim.org/>`__ and `Emacs <https://www.gnu.org/software/emacs/>`__, have the advantage of being found on practically every UNIX system and not requiring a graphical desktop, which means they can be used when logging into a machine remotely via SSH, or on older machines that may feel sluggish when running some of the graphical editors we discuss below. They can also be extremely powerful, and can be customized to work with pretty much any programming language under the sun.
+
+Even if you don't use a terminal-based editor as your primary development environment, we recommend building at least some basic familiarity with either vim or Emacs, in case you are ever in a situation where you can only edit a file through a terminal (and can't launch a graphical editor).
+
+
+Visual Studio Code
+~~~~~~~~~~~~~~~~~~
+
+If you want to use a graphical text editor (i.e., one that uses a graphical user interface from the Linux desktop), a popular option is `Visual Studio Code <https://code.visualstudio.com/>`__ (not to be confused with its older sibling, `Visual Studio <https://visualstudio.microsoft.com/vs/>`__). It is more powerful than a regular text editor, with features like syntax highlighting, auto completion, plugins that integrate with other tools, etc. but it is not a full-fledged `Integrated Developent Environment <https://en.wikipedia.org/wiki/Integrated_development_environment>`__ (and, as such, is a more lightweight piece of software requiring less resources).
```

Comment:
It may be helpful to specifically point out VS Code's has really great integration with git, I've found a lot of people don't use it and it has saved me a ton of time.

Reply:
Added mention of the git integration.